crates/re_viewer_context/src/space_view/scene.rs

```diff
@@ -18,17 +18,24 @@ pub trait Scene {
         query: &SceneQuery<'_>,
         space_view_state: &dyn SpaceViewState,
         highlights: SpaceViewHighlights,
-    ) -> Vec<re_renderer::QueueableDrawData>;
+    );
 
-    /// Converts itself to a reference of [`std::any::Any`], which enables downcasting to concrete types.
-    fn as_any(&self) -> &dyn std::any::Any;
+    /// Converts itself to a mutable reference of [`std::any::Any`], which enables downcasting to concrete types.
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any;
 }
 
 /// Implementation of [`Scene`] for a specific [`SpaceViewClassImpl`].
 pub struct TypedScene<C: SpaceViewClassImpl> {
     pub context: C::SceneContext,
     pub parts: C::ScenePartCollection,
     pub highlights: SpaceViewHighlights,
+
+    /// All draw data gathered during the last call to [`Self::populate`].
+    ///
+    /// TODO(wumpf): Right now the ui methods control when and how to create [`re_renderer::ViewBuilder`]s.
+    ///              In the future, we likely want to move view builder handling to `re_viewport` with
+    ///              minimal configuration options exposed via [`crate::SpaceViewClass`].
+    pub draw_data: Vec<re_renderer::QueueableDrawData>,
```

Comment:
I realize we *also* expose QueableDrawData via the `populate` interface, but having it visible here feels weird to me, especially given that not all Scenes actually need to use the re_renderer at all. Could we not pass all the QueueableDrawData around using a `RenderContext` stored as a member of `SpatialSceneContext` (and other views that make use of the renderer).

Reply:
we'd need to add a known render context, i.e. generating the same problem as the entity highlights.
As described by the todo I'd like to live in a world where we always have a viewbuilder that is controlled by the framework if any of the scene parts returned a renderable. That _does_ have some drawbacks though: It makes it much harder to build UI _around_ your viewbuilder as you're probably forced to have the entire egui_tile tile filled out by re_renderer.